tower_cli/cli/misc.py

```diff
@@ -228,19 +230,37 @@ def login(username, password, scope, verbose):
 
     # Explicitly set a basic auth header for PAT acquisition (so that we don't
     # try to auth w/ an existing user+pass or oauth2 token in a config file)
+
     req = collections.namedtuple('req', 'headers')({})
-    HTTPBasicAuth(username, password)(req)
-    r = client.post(
-        '/users/{}/personal_tokens/'.format(username),
-        data={"description": "Tower CLI", "application": None, "scope": scope},
-        headers=req.headers
-    )
+    if client_id and client_secret:
```

Comment:
We should also have an `elif` here for if only the `client_id` flag is provided.  In this case, the grant type would be "public".  For example:

```{
    "name": "Public App",
    "description": "for the people",
    "user": 1,
    "client_type": "public",
    "redirect_uris": "",
    "authorization_grant_type": "password",
    "skip_authorization": false
}
```

Reply:
Good point - addressed!

Reply:
LGTM.  Thanks for adding that.